tests/utilities/test_attributedict.py

```diff
@@ -74,3 +74,8 @@ def test_attributedict_inequality(dict1, dict2):
     assert AttributeDict(dict1) != dict2
     assert AttributeDict(dict1) != AttributeDict(dict2)
     assert dict1 != AttributeDict(dict2)
+
+
+def test_attributedict_recursive():
```

Comment:
I think this warrants an expanded test suite now that I've seen the implementation.  Try to cover the following.

* nested dicts
* lists and tuples with dicts
* dicts with lists with dicts
* stuff with integers, bools, strings
* something with a set.